Client/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -42,6 +44,58 @@ extension BrowserViewController: TopToolbarDelegate {
     present(container, animated: true)
   }
 
+  func topToolbarDidPressLockImageView(_ urlBar: TopToolbarView) {
+    guard let webView = tabManager.selectedTab?.webView else {
+      log.error("Invalid WebView")
+      return
+    }
+    
+    let getServerTrustForErrorPage = { () -> SecTrust? in
+      do {
+        if let url = webView.url {
+          return try ErrorPageHelper.serverTrust(from: url)
+        }
+      } catch {
+        log.error(error)
+      }
+      
+      return nil
+    }
+    
+    guard let trust = webView.serverTrust ?? getServerTrustForErrorPage() else {
+      return
+    }
+
+    let serverCertificates = Array(
+      (0..<SecTrustGetCertificateCount(trust))
+        .compactMap { SecTrustGetCertificateAtIndex(trust, $0) })  // Should be `OrderedSet`
+
+    // TODO: Instead of showing only the first cert in the chain,
+    // have a UI that allows users to select any certificate in the chain (similar to Desktop browsers)
```

Comment:
Maybe add an issue for this to track for further improvement in the future and then ref it here to find it later

Reply:
https://github.com/brave/brave-ios/issues/5229